src/api/src/http/resource.rs

```diff
@@ -59,27 +53,135 @@ pub(crate) async fn get_resource(
 
     info!("Resource description: {:?}", &resource_description);
 
-    if session.tee_public_key().is_none() {
-        error!("TEE pubkey not found");
-        raise_error!(Error::FailedAuthentication(String::from(
-            "Teepub key not found in session"
-        )));
-    }
-
     let resource_byte = repository
         .read()
         .await
         .read_secret_resource(resource_description)
         .await
         .map_err(|e| Error::ReadSecretFailed(e.to_string()))?;
 
-    let jwe = session
-        .to_jwe(resource_byte)
-        .map_err(|e| Error::JWEFailed(e.to_string()))?;
+    let jwe = jwe(pubkey, resource_byte)?;
 
     let res = serde_json::to_string(&jwe).map_err(|e| Error::JWEFailed(e.to_string()))?;
 
     Ok(HttpResponse::Ok()
         .content_type("application/json")
         .body(res))
 }
+
+async fn get_pubkey_from_cookie(
+    request: &HttpRequest,
+    map: web::Data<SessionMap<'_>>,
+) -> Result<TeePubKey> {
+    // check cookie
+    let cookie = request
+        .cookie(KBS_SESSION_ID)
+        .ok_or(Error::UnAuthenticatedCookie)?;
+
+    let session_map = map.sessions.read().await;
+    let locked_session = session_map
+        .get(cookie.value())
+        .ok_or(Error::UnAuthenticatedCookie)?;
+
+    let session = locked_session.lock().await;
+
+    info!("Cookie {} request to get resource", session.id());
+
+    if !session.is_authenticated() {
+        error!("UnAuthenticated KBS cookie {}", cookie.value());
+        raise_error!(Error::UnAuthenticatedCookie);
+    }
+
+    if session.is_expired() {
+        error!("Expired KBS cookie {}", cookie.value());
+        raise_error!(Error::ExpiredCookie);
+    }
+
+    let pubkey = session.tee_public_key().ok_or(Error::PublicKeyGetFailed(
+        "No public key in the session".into(),
+    ))?;
+
+    Ok(pubkey)
+}
+
+async fn get_pubkey_from_header(
+    request: &HttpRequest,
+    token_broker: web::Data<Arc<RwLock<dyn AttestationTokenBroker + Send + Sync>>>,
+) -> Result<TeePubKey> {
+    let bearer = Authorization::<Bearer>::parse(request)
+        .map_err(|e| Error::InvalidRequest(format!("parse Authorization header failed: {e}")))?
+        .into_scheme();
+
+    let token = bearer.token().to_string();
+
+    let claims = token_broker
+        .read()
+        .await
+        .verify(token)
```

Comment:
We'd better verify token here locally instead of call `token_broker` to do it, because the issuer may not be this KBS itself. I think we  can set a "get_token_verification_pubkey" API, it can help to get pubkey from  token broker or remote token issuer, which can be specified by config parameter.

Reply:
Currently we only suppose this token is provisioned by the same KBS just then, for the token is short-lived ~10min. Do you mean a long-lived token?

Reply:
This is not related to the lifecycle of the token. According to my understanding, in the confidential data hub design, the issuer of the token will not be the same entity as the consumer of the token most of the time, even if it is a token with a short lifecycle, it does not matter.

Reply:
Do you mean a scenario like this:
1. Client gets a token provisioned by an attestation service rather than CoCo. For example Amber
2. Client wants to use this token to request resource in KBS

Reply:
> Do you mean a scenario like this:
> 
> 1. Client gets a token provisioned by an attestation service rather than CoCo. For example Amber
> 2. Client wants to use this token to request resource in KBS

Exactly. The token issuer in 1 may not only be Amber, but also CoCo AS deployed by trusted parties (specifically used to issue tokens).

Reply:
Ok. I think this should be carefully handled. I think there are two different tokens in the context:
I. In KBS: Signed(claims, pubkey)
II. In Amber: Signed(claims)

We should firstly ensure this token is with the proper token (type I), with which we can encrypt the dataflow of the confidential resource. If the token is type II, we should think about how we send back the confidential resource.

Also, the basic logic of a certificated token is:
1. Verify the endorsement and claims due to the user defined policy
3. A token is provisioned

Also, if tokens provisioned by other CoCo-KBS is to be supported, we should recheck the claims via giving it to the backend CoCo-AS against the policy of the current CoCo-KBS/AS. 

My idea is that we can start with a simple scenario, s.t.
1. This token is provisioned by the same KBS.

To accept a token from another KBS, we should
1. impl new API for AS to receive different signed claims rather than a raw evidence to verify against a policy.
2. Let KBS leverage this new API to forward the signed claims of the token (or directly the token)

To accept a token from another attestation service like Amber, we should at least
1. Let Amber to support a token with pubkey
Or 2. Think of a way where a verified token can be derived into a tee key pair to fit in the KBS protocol to encrypt the dataflow
Or 3. Do some modifications upon the current KBS protocol

Reply:
OK, I agree start with the simple scenario that token provisioned by the same KBS. 

However, two points need to be emphasized:

1. Verify that the claims in the token should not be in AS, but should introduce a new policy engine in KBS, which is consistent with the RATS passport model. The policy here is the policy of rely party verifying the attachment results.
2. After this PR is merged, I will transfer the logic of the Token broker from KBS to AS. In other words, AS no longer returns a results claims, but directly returns the token. Therefore, I may modify the token's verification logic here to: obtain the token verification public key from AS's new API and then have KBS verify the token.